libs/sdk-ui-tests-e2e/cypress/tools/dateFilter.ts

```diff
@@ -51,4 +82,35 @@ export class DateFilter {
     timeSwitcherHasValue(value: string): Cypress.Chainable {
         return cy.get(".time-switcher").should("have.text", value);
     }
+
+    hasRelativePreset(preset: RelativePreset, expect = true) {
+        this.getOption(`.s-relative-preset-${preset}`).should(expect ? "exist" : "not.exist");
+        return this;
+    }
+
+    hasAbsolutePreset(preset: AbsolutePreset, expect = true) {
+        this.getOption(`.s-absolute-preset-${preset}`).should(expect ? "exist" : "not.exist");
+        return this;
+    }
+
+    apply() {
+        this.getDropdownElement().find(".s-date-filter-apply").click({ scrollBehavior: false });
+        return this;
+    }
+
+    applyEnabled(expect = true) {
+        this.getDropdownElement()
+            .find(".s-date-filter-apply")
+            .should(expect ? "not.have.class" : "have.class", "disabled");
+        return this;
+    }
+
+    cancel() {
+        this.getDropdownElement().find(".s-date-filter-cancel").click({ scrollBehavior: false });
+        return this;
+    }
+
+    private getOption(selector: string) {
```

Comment:
do we need a private method here?

Reply:
getOption is only used in DateFilter, so it need to be a private method. why should we make it as a public method while it is not used outside class?

Reply:
As I see, there is only `getOption()`method is used as private. how about others? I think `cancel()` or `applyEnable() `also use in this class, no more outside

Reply:
cancel() and applyEnable() have not been used yet, they are created with purpose to use them in .spec file, so we define them as public method
getOption() is only created with purpose to use in only class dateFilter.ts, for Ex: selectRelativePreset, selectRelativeForm will call getOption().

Reply:
Agree